### DISCUSSION TITLE
I suggest this.

### DISCUSSION THREAD

**Discussion by hebaek76916** [Yummypets/YPImagePicker#695] [Vote for Issue to faster implement.]
More Instagram like,
`YPLibraryVC+CollectionView.swift`

`

    func deselect(indexPath: IndexPath) {

        if let positionIndex = selection.firstIndex(where: {
			$0.assetIdentifier == mediaManager.fetchResult[indexPath.row].localIdentifier
		}) {

            selection.remove(at: positionIndex)
            // Refresh the numbers
            var selectedIndexPaths = [IndexPath]()
            mediaManager.fetchResult.enumerateObjects { [unowned self] (asset, index, _) in
                if self.selection.contains(where: { $0.assetIdentifier == asset.localIdentifier }) {
                    selectedIndexPaths.append(IndexPath(row: index, section: 0))
                }
            }
            v.collectionView.reloadItems(at: selectedIndexPaths)
            
            var selectionLastIndex: IndexPath?
            if selection.count != 0 { // 🕊 ~
                let last = selection.last!.index
                selectionLastIndex = IndexPath(row: last, section: 0)
                v.collectionView.deselectItem(at: indexPath, animated: false)
                v.collectionView.selectItem(at: selectionLastIndex, animated: false, scrollPosition: [])
                currentlySelectedIndex = selectionLastIndex!.row
                changeAsset(mediaManager.fetchResult[selectionLastIndex!.row]) //  ~🕊
            } else { // 🕊
         
                // Replace the current selected image with the previously selected one
                if let previouslySelectedIndexPath = selectedIndexPaths.last {
                    v.collectionView.deselectItem(at: indexPath, animated: false)
                    v.collectionView.selectItem(at: previouslySelectedIndexPath, animated: false, scrollPosition: [])
                    currentlySelectedIndex = previouslySelectedIndexPath.row
                    changeAsset(mediaManager.fetchResult[previouslySelectedIndexPath.row])
                }
            } // 🕊
            checkLimit()
        }
    }
`

I think when deselect, move to `selection`'s last index is more make sense.

How about my opinion?..

**Comment by NikKovIos** [comment]
For now it act like in instagram when you select or deselect in multiple selection mode.
